pkg/asset/manifests/powervs/cluster.go

```diff
@@ -0,0 +1,238 @@
+	}
+
+	transitGatewayName = fmt.Sprintf("%s-tg", clusterID.InfraID)
+
+	cosName = fmt.Sprintf("%s-cos", clusterID.InfraID)
+
+	switch {
+	case installConfig.Config.Platform.PowerVS.VPCRegion != "":
+		cosRegion, err = powervstypes.COSRegionForVPCRegion(installConfig.Config.Platform.PowerVS.VPCRegion)
+		if err != nil {
+			return nil, fmt.Errorf("GenerateClusterAssets: COSRegionForVPCRegion returns %w", err)
+		}
+	case installConfig.Config.Platform.PowerVS.Region != "":
+		cosRegion = installConfig.Config.Platform.PowerVS.Region
+	default:
+		return nil, fmt.Errorf("GenerateClusterAssets: Region is empty")
+	}
+
+	imageName = fmt.Sprintf("rhcos-%s", clusterID.InfraID)
+
+	bucketName = fmt.Sprintf("%s-bootstrap-ign", clusterID.InfraID)
+
+	powerVSCluster = &capibm.IBMPowerVSCluster{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: capibm.GroupVersion.String(),
+			Kind:       "IBMPowerVSCluster",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterID.InfraID,
+			Namespace: capiutils.Namespace,
+			Annotations: map[string]string{
+				"powervs.cluster.x-k8s.io/create-infra": "true",
+			},
+		},
+		Spec: capibm.IBMPowerVSClusterSpec{
+			Network: capibm.IBMPowerVSResourceReference{
+				Name: &network,
+			},
+			ServiceInstance: &service,
+			Zone:            &installConfig.Config.Platform.PowerVS.Zone,
+			ResourceGroup: &capibm.IBMPowerVSResourceReference{
+				Name: &resourceGroupName,
+			},
+			VPC: &capibm.VPCResourceReference{
+				Name:   &vpcName,
+				Region: &vpcRegion,
+			},
+			TransitGateway: &capibm.TransitGateway{
+				Name: &transitGatewayName,
+			},
+			LoadBalancers: []capibm.VPCLoadBalancerSpec{
+				{
+					Name:   fmt.Sprintf("%s-loadbalancer", clusterID.InfraID),
+					Public: ptr.To(true),
+					AdditionalListeners: []capibm.AdditionalListenerSpec{
+						{
+							Port: 22,
+						},
+						// @BUG We should be able to specify this:
+						// capibm.AdditionalListenerSpec{
+						//	Port: 6443,
+						// },
+					},
+				},
+				{
+					Name:   fmt.Sprintf("%s-loadbalancer-int", clusterID.InfraID),
+					Public: ptr.To(false),
+					AdditionalListeners: []capibm.AdditionalListenerSpec{
+						// @BUG We should be able to specify this:
+						// capibm.AdditionalListenerSpec{
+						//	Port: 6443,
+						// },
+						{
+							Port: 22623,
+						},
+					},
+				},
+			},
+			CosInstance: &capibm.CosInstance{
+				Name:         cosName,
+				BucketName:   bucketName,
+				BucketRegion: vpcRegion,
+			},
+			Ignition: &capibm.Ignition{
+				Version: "3.4",
+			},
+		},
+	}
+
+	// Avoid:
+	//   vpc:
+	//     name: ""
+	//     region: ""
+	if installConfig.Config.Platform.PowerVS.VPCName != "" {
+		vpcResource := capibm.VPCResourceReference{
+			Name:   &installConfig.Config.Platform.PowerVS.VPCName,
+			Region: &installConfig.Config.Platform.PowerVS.VPCRegion,
+		}
+		reflect.ValueOf(&powerVSCluster.Spec).Elem().FieldByName("VPC").Set(reflect.ValueOf(&vpcResource))
+	}
```

Comment:
```suggestion

```
With the changes above to set the `vpcName`, this shouldn't be needed?

Reply:
Even that code wasn't working correctly when I tested it  This worked:
```
        logrus.Debugf("GenerateClusterAssets: len(VPCSubnets) = %d", len(installConfig.Config.Platform.PowerVS.VPCSubnets))
        if len(installConfig.Config.Platform.PowerVS.VPCSubnets) > 0 {
                powerVSCluster.Spec.VPCSubnets = make([]capibm.Subnet, 0)
                for i, _ := range installConfig.Config.Platform.PowerVS.VPCSubnets {
                        powerVSCluster.Spec.VPCSubnets = append(powerVSCluster.Spec.VPCSubnets,
                                capibm.Subnet{
                                        ID: &installConfig.Config.Platform.PowerVS.VPCSubnets[i],
                                })
                }
                logrus.Debugf("GenerateClusterAssets: subnets = %+v", powerVSCluster.Spec.VPCSubnets)
        }
```

Reply:
Yes, it is not needed